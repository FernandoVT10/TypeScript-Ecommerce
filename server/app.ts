import express from "express";
import routes from "./routes";
import bodyParser from "body-parser";
import cookieParser from "cookie-parser";

const app = express();

app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));

app.use(cookieParser());

routes(app);

export default app;
