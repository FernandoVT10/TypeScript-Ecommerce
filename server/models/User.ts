import mongoose, { Schema, Document } from "mongoose";

const EMAIL_VALIDATION_REGEX = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;

export interface IUser extends Document {
    name: string,
    username: string,
    email: string,
    password: string,
    active?: boolean,
    activeToken: string
}

const userSchema = new Schema({
    name: {
        type: String,
	maxlength: 200,
        required: true
    },
    username: {
	type: String,
	maxlength: 30,
	required: true,
	unique: true
    },
    email: {
	type: String,
	validate: {
	    validator: (email: string) => EMAIL_VALIDATION_REGEX.test(email),
	    message: "The email is invalid"
	},
	required: true,
	unique: true
    },
    password: {
	type: String,
	required: true
    },
    active: {
	type: Boolean,
	default: false
    },
    activeToken: String
}, { timestamps: true });

export default mongoose.model<IUser>("users", userSchema);
