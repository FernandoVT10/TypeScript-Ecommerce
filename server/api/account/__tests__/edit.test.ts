import app from "../../../app";
import supertest from "supertest";

import User from "../../../models/User";

const request = supertest(app);

setupTestDB("test_account_edit_api");
mockAuthentication();

describe("api/account/edit", () => {
    it("should edit a user correcty", async () => {
	const res = await request.put("/api/account/edit")
	    .send({
	    	name: "edited name",
		username: "edited username",
		email: "test@edited.com"
	    })
	    .set("Authorization", "Bearer token");

	const user = await User.findOne();

	expect(user.name).toBe("edited name");
	expect(user.username).toBe("edited username");
	expect(user.email).toBe("test@edited.com");

	expect(res.body.data.message).toBe("The account has been edited successfully");
    });

    it("should return an error when the values are invalid", async () => {
	const res = await request.put("/api/account/edit")
	    .send({
	    	name: "edited name",
		username: "edited username",
		email: "test@invalid"
	    })
	    .set("Authorization", "Bearer token");

	expect(res.body).toMatchSnapshot();
    });
});
