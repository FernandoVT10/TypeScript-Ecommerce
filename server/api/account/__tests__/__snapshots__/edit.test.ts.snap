// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/account/edit should return an error when the values are invalid 1`] = `
Object {
  "error": "Bad Request",
  "message": "users validation failed: email: The email is invalid",
  "path": "/api/account/edit",
  "status": 400,
}
`;
